mcp_server_gravitino/server/tools/tag.py

```diff
@@ -62,96 +68,28 @@ def associate_tag_to_table(mcp: FastMCP, session: httpx.Client) -> None:
-        dict[str, str]
-            If an error occurs, returns {"result": "error", "message": "error message"}.
-            if successful, returns {"result": "success"}
-        """
+    def _associate_tag_to_entity(
+        tag_name: str,
+        fully_qualified_name: str,
+    ) -> dict[str, str]:
         if not fully_qualified_name:
             return {"result": "error", "message": "fully_qualified_name cannot be empty"}
-        table_names = fully_qualified_name.split(".")
-        if len(table_names) != 4:
-            return {
-                "result": "error",
-                "message": "table fully_qualified_name should be in the format 'metalake.catalog.schema.table'",
-            }
-
-        # metalake = table_names[0]
-        catalog_name = table_names[1]
-        schema_name = table_names[2]
-        table_name = table_names[3]
-
-        qualified_name = f"{catalog_name}.{schema_name}.{table_name}"
 
-        return _associate_tag_to_object(
-            session=session, tag_name=tag_name, object_type="table", obj_qualified_name=qualified_name
-        )
-
-
-def associate_tag_to_column(mcp: FastMCP, session: httpx.Client) -> None:
-    @mcp.tool(
-        name="associate_tag_to_column",
-        description="associate tag to a column object",
-        tags={
-            TAG_OBJECT_TAG,
-        },
-        annotations={
-            "readOnlyHint": False,
-            "openWorldHint": True,
-            "destructiveHint": True,
-            "idempotentHint": True,
-        },
-    )
-    def _associate_tag_to_column(tag_name: str, table_fully_qualified_name: str, column_name: str) -> dict[str, str]:
-        """
-        Associate a tag with a column by tag name, table's fully qualified name and column name.
-
-        Parameters
-        ----------
-        tag_name : str
-            The name of the tag to be associated with the column
-        table_fully_qualified_name : str
-            Fully qualified name of the table
-        column_name : str
-            Name of the column
-
-        Returns
-        -------
-        dict[str, str]
-            If an error occurs, returns {"result": "error", "message": "error message"}.
-            if successful, returns {"result": "success"}
-        """
-        if not table_fully_qualified_name:
-            return {"result": "error", "message": "table_fully_qualified_name cannot be empty"}
-        table_names = table_fully_qualified_name.split(".")
-        if len(table_names) != 4:
+        names = fully_qualified_name.split(".")
+        if len(names) not in [4, 5]:
             return {
                 "result": "error",
-                "message": "table_fully_qualified_name should be in the format 'metalake.catalog.schema.table'",
+                "message": "fully_qualified_name should be in the format'metalake.catalog.schema.table' or 'metalake.catalog.schema.table.column'",
             }
+        object_type = "table" if len(names) == 4 else "column"
```

Comment:
this logic can be enhanced to support catalog and schema as well, depends on the length of the fqn name.